src/cwe_checker_lib/src/intermediate_representation/project.rs

```diff
@@ -249,21 +249,68 @@ mod tests {
     use super::*;
 
     impl Project {
-        pub fn mock_empty() -> Project {
-            let register_set = vec!["RAX", "RCX", "RDX", "RBX", "RSP", "RBP", "RSI", "RDI"]
-                .into_iter()
-                .map(|name| Variable::mock(name, ByteSize::new(8)))
-                .collect();
+        /// Returns project with x64 calling convention and mocked program.
+        pub fn mock_x64() -> Project {
+            let mut none_cconv_register: Vec<Variable> = vec![
+                "RAX", "RBX", "RSP", "RBP", "R10", "R11", "R12", "R13", "R14", "R15",
+            ]
+            .into_iter()
+            .map(|name| Variable::mock(name, ByteSize::new(8)))
+            .collect();
+            let mut integer_register = CallingConvention::mock_x64().integer_parameter_register;
+            integer_register.append(&mut none_cconv_register);
+
+            let calling_conventions: BTreeMap<String, CallingConvention> =
+                BTreeMap::from([("__stdcall".to_string(), CallingConvention::mock_x64())]);
+
             Project {
                 program: Term {
                     tid: Tid::new("program_tid"),
-                    term: Program::mock_empty(),
+                    term: Program::mock(),
                 },
                 cpu_architecture: "x86_64".to_string(),
                 stack_pointer_register: Variable::mock("RSP", 8u64),
-                calling_conventions: BTreeMap::new(),
-                register_set,
-                datatype_properties: DatatypeProperties::mock(),
+                calling_conventions,
+                register_set: integer_register.iter().cloned().collect(),
+                datatype_properties: DatatypeProperties::mock_x64(),
+            }
+        }
+
+        pub fn mock_arm32() -> Project {
+            let none_cconv_4byte_register: Vec<Variable> = vec!["r12", "r14", "r15"]
+                .into_iter()
+                .map(|name| Variable::mock(name, ByteSize::new(4)))
+                .collect();
+
+            let none_cconv_16byte_register: Vec<Variable> = vec![
+                "q0", "q1", "q2", "q3", "q8", "q9", "q10", "q11", "q12", "q13", "q14", "q15",
+            ]
+            .into_iter()
+            .map(|name| Variable::mock(name, ByteSize::new(16)))
+            .collect();
+
+            let callee_saved_register = CallingConvention::mock_arm32().callee_saved_register;
+
+            let integer_register = CallingConvention::mock_arm32()
+                .integer_parameter_register
+                .into_iter()
+                .chain(none_cconv_4byte_register)
+                .chain(none_cconv_16byte_register)
+                .chain(callee_saved_register);
+
+            Project {
+                program: Term {
+                    tid: Tid::new("program_tid"),
+                    term: Program::mock(),
+                },
+                cpu_architecture: "arm32".to_string(),
+                stack_pointer_register: Variable::mock("r13", 4u64),
```

Comment:
I think Ghidra names the stack pointer register `sp` instead of `r13`. Can you double-check and if necessary change this?